File: test-suite/src/test/java/io/micronaut/docs/informer/SharedInformerCache.java
package io.micronaut.docs.informer;

import io.kubernetes.client.informer.SharedIndexInformer;
import io.kubernetes.client.informer.cache.Indexer;
import io.kubernetes.client.openapi.models.V1ConfigMap;
import io.micronaut.context.annotation.Requires;
import io.micronaut.kubernetes.client.informer.SharedIndexInformerFactory;
import jakarta.inject.Singleton;

import java.util.List;

@Requires(property = "spec.name", value = "ConfigMapInformerSpec")
//tag::cache[]
@Singleton
public class SharedInformerCache {

    private final SharedIndexInformerFactory sharedIndexInformerFactory;

    public SharedInformerCache(SharedIndexInformerFactory sharedIndexInformerFactory) {
        this.sharedIndexInformerFactory = sharedIndexInformerFactory;
    }

    /**
     * Get all config maps from informer from namespace.
     */
    List<V1ConfigMap> getConfigMaps(String namespace) {
        SharedIndexInformer<V1ConfigMap> sharedIndexInformer = sharedIndexInformerFactory.getExistingSharedIndexInformer(namespace, V1ConfigMap.class);
        if (sharedIndexInformer != null) {
            Indexer<V1ConfigMap> indexer = sharedIndexInformer.getIndexer();
            return indexer.list();
        } else {
            return null;
        }
    }
}
//end::cache[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/informer/SharedInformerCache.kt
package io.micronaut.docs.informer

import io.kubernetes.client.openapi.models.V1ConfigMap
import io.micronaut.context.annotation.Requires
import io.micronaut.kubernetes.client.informer.SharedIndexInformerFactory
import jakarta.inject.Singleton

@Requires(property = "spec.name", value = "ConfigMapInformerSpec")
//tag::cache[]
@Singleton
class SharedInformerCache(private val sharedIndexInformerFactory: SharedIndexInformerFactory) {

    /**
     * Get all config maps from informer from namespace.
     */
    fun getConfigMaps(namespace: String): List<V1ConfigMap>? {
        val sharedIndexInformer = sharedIndexInformerFactory.getExistingSharedIndexInformer(namespace, V1ConfigMap::class.java)
        return if (sharedIndexInformer != null) {
            val indexer = sharedIndexInformer.indexer
            indexer.list()
        } else {
            null
        }
    }
}
//end::cache[]
